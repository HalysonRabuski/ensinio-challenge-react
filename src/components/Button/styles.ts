import styled from 'styled-components'

interface ButtonProps {
  padding: string
  width: string
  color: string
  bgColor: string
  border: string
}

export const Button = styled.a<ButtonProps>`
  width: ${(props) => props.width};
  padding: ${(props) => props.padding};
  background-color: ${(props) => props.bgColor};
  color: ${(props) => props.color};
  border: ${(props) => props.border};
  border-radius: 80px;
  text-align: center;
  &:hover {
    cursor: pointer;
  }
`
