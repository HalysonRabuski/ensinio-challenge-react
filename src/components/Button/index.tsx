import React, { PropsWithChildren } from 'react'
import { Button } from './styles'

interface ButtonProps {
  width?: string
  padding?: string
  color?: string
  bgColor?: string
  border?: string
}

const TopMenu: React.FC<PropsWithChildren<ButtonProps>> = ({
  padding = '5px',
  width = '160px',
  color = 'white',
  bgColor = 'none',
  border = '1px solid',
  children,
}) => (
  <Button
    data-testid="button-id"
    width={width}
    padding={padding}
    color={color}
    bgColor={bgColor}
    border={border}
  >
    {children}
  </Button>
)

export default TopMenu
