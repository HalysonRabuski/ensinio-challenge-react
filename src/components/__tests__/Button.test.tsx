/* eslint-disable react/jsx-filename-extension */
import React from 'react'
import { render, screen, cleanup } from '@testing-library/react'
import renderer from 'react-test-renderer'
// import '../../i18n'
// import renderer from 'react-test-renderer'
import Button from '../Button'

afterEach(() => {
  cleanup()
})

jest.mock('react-i18next', () => ({
  // this mock makes sure any components using the translate hook can use it without a warning being shown
  useTranslation: () => ({
    t: (str: any) => str,
    i18n: {
      changeLanguage: () => new Promise(() => {}),
    },
  }),
}))

test('should render card component', async () => {
  render(<Button />)
  const ButtonElement = screen.getByTestId('button-id')

  expect(ButtonElement).toBeInTheDocument()
})

test('should render card component with an image and texts as children', () => {
  const ButtonElement = renderer.create(<Button />).toJSON()
  expect(ButtonElement).toMatchSnapshot()
})
