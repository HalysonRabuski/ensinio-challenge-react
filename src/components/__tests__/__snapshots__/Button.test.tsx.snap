// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render card component with an image and texts as children 1`] = `
<a
  className="sc-bdnxRM lkgSmQ"
  color="white"
  data-testid="button-id"
  width="160px"
/>
`;
